rtos/Kernel.cpp

```diff
@@ -0,0 +1,52 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+#include "rtos/Kernel.h"
+
+#include "cmsis_os2.h"
+
+namespace rtos {
+
+uint64_t Kernel::get_ms_count() {
+    return osKernelGetTickCount();
+}
+
+osStatus Kernel::delay(uint32_t millisec) {
+    return osDelay(millisec);
+}
+
+osStatus Kernel::delay_until(uint64_t millisec) {
+    return osDelayUntil(millisec);
+}
+
+int32_t Kernel::lock() {
```

Comment:
Do you have any use cases for lock and unlock in mind? I would be inclined to leave this private unless it is needed for a certain reason.

Reply:
No immediate use, but it seemed like if adding a Kernel class, it was the most obvious API from osKernel that you people might want. Otherwise it will just be the tick count, if I remove the delays, as I'm now minded to.